### ISSUE TITLE
[3.0] Hotkey CTRL+DOWN does not perform default pull button action anymore

### ISSUE THREAD

**Issue by drauch** [gitextensions/gitextensions#5885]
**What is the current behavior?**

In 2.x the CTRL+DOWN hotkey performed the default pull button action. In 3.0 the behavior changed, it always opens the fetch/pull dialog, regardless of what I choose as the default pull button action.

**What is the expected behavior?**

Is there a new hotkey? I can't find one. I want to do "fetch all" without any dialogs inbetween with a simple hotkey.

**Comment by vbjay** [comment]
![image](https://user-images.githubusercontent.com/3629489/49806283-56ff7680-fd25-11e8-9899-983ccda76f41.png)

**Comment by vbjay** [comment]
We have an issue template for a reason.  
```
<!--
IMPORTANT

Before creating an issue, be sure to search existing issues, both open AND closed,
to see whether someone else has already reported your issue.

Please also read CONTRIBUTING.md.
-->

**Do you want to request a *feature* or report a *bug*?**

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**

**Environment you encounter the issue:**

<!--
Help->About Copy button or Help->Report an issue link will put this info in the clipboard.
-->
- GitExtensions version:
- GIT version: 
- OS version: 
- .NET version: 

**Did this work in previous version of GitExtensions (which)?**

<!-- Love gitextensions? Please consider supporting our collective:
👉  https://opencollective.com/gitextensions/donate -->
```

**Comment by drauch** [comment]
Quick fetch is not what I want. It does "fetch" and not "fetch all", regardless of what setting I set as the default.

Also, I used the issue template, just deleted what was irrelevant. If you need more information, please say so, I'm happy to provide it.

**Comment by vbjay** [comment]
Always include environment info. Help-> About will let you copy this from GE.   

We have an issue template for a reason. It gives us what we need. We expect the issue template to be filled out.

**Comment by vbjay** [comment]
So add a script that does what you want and set the hotkey to the desired
value.

**Comment by drauch** [comment]
- Git Extensions 3.00.00.4433
- Build fca7cf228b481ee8c1b779cf7b882ccdfbdcd1bc
- Git 2.20.0.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3221.0
- DPI 96dpi (no scaling)

> So add a script that does what you want and set the hotkey to the desired value.

How do you insert "perform default pull button action" into a script?

**Comment by vbjay** [comment]
You script the actual git command.   If you want fetch all or even fetch all with submodules then
Command git
```bash
fetch --all
```
Or 
Command sh
```bash
-c " git fetch --all && git submodule foreach --recursuve git fetch --all"
```

**Comment by drauch** [comment]
But I do not want to _always_ use fetch all, just in some repositories. I want one hotkey for "default pull button action", just like in 2.x :-)

**Comment by gerhardol** [comment]
This seem to be hardcoded now, probably a consequence of the rewritten default handling
```            this.pullToolStripMenuItem.ShortcutKeys = ((System.Windows.Forms.Keys)((System.Windows.Forms.Keys.Control | System.Windows.Forms.Keys.Down)));
```

But the workaround is not too difficult: Press return, the form is filled in correctly

**Comment by drauch** [comment]
The workaround is unpleasant, opening the dialog takes up to five seconds. I guess this is a performance problem...

...would be glad if the shortcut could be configured just like in the 2.x version 👍

**Comment by gerhardol** [comment]
> ...would be glad if the shortcut could be configured just like in the 2.x version 

The shortcut was hardcoded in 2.x, just that the Form did not open

**Comment by drauch** [comment]
> just that the Form did not open
In that case, I'd love to have that again :)

**Comment by vbjay** [comment]
It takes roughly a second maybe less for the form to show.  Record how long it really takes use https://www.screentogif.com/  trim the beginning before you pressed the keys.  Drag created gif to a comment here.

**Comment by gerhardol** [comment]
There are over 20 hardcoded shortcuts that should be made configurable.
The hardcoded shortcuts could conflict with language defaults.

In this case the changed functionality for the pull dialog (the "set next action as default") that likely is the reason.

**Comment by vbjay** [comment]
Yep. Wouldn't want a fetch and prune with a simple key press.  Pretty sure that's why the shortcut shows the dialog.

**Comment by drauch** [comment]
@vbjay : I press the shortcut CTRL+DOWN immediately after opening the repository. That's when it takes so long. Is this scenario faster for you? I have made a screen recording with ScreenToGif but realized that I can't make it public for legal reasons. If you give me your e-mail address I can send it in private to you if you like.

**Comment by vbjay** [comment]
I tend to wait for the form to be up and it is done doing its thing.

That's ok. I was mainly looking for timing. Does it only do it for your
repo or every repo?
